### ISSUE TITLE
processor.py 中 padding函数里为什么要对 sample按照 length排序

### ISSUE THREAD

**Issue by CodeMonkZy** [wenet-e2e/wenet#1850]
部分代码：
        order = torch.argsort(feats_length, descending=True)
        feats_lengths = torch.tensor(
            [sample[i]['feat'].size(0) for i in order], dtype=torch.int32)
        sorted_feats = [sample[i]['feat'] for i in order]
        sorted_keys = [sample[i]['key'] for i in order]
        sorted_labels = [
            torch.tensor(sample[i]['label'], dtype=torch.int64) for i in order
        ]
到这里sample已经是一个batch的数据，代码里在padding前对batch里的数据按照length排序，然后再pad，这个排序的目的是什么？感谢解答！

**Comment by robin1001** [comment]
这是个历史原因，因为后面要调用 pad_sequence 做 padding，我记很早之前 pad_sequence 要求输入序列排序，但现在我看了一下，没有这个限制了。详见：

https://pytorch.org/docs/stable/generated/torch.nn.utils.rnn.pad_sequence.html

一个 batch 内排不排序不影响最终计算的 loss 和梯度，对最终训练结果也不影响。
